cypress/integration/loggedOutFlows/runtimeBannerDeepLinking.spec.js

```diff
@@ -0,0 +1,76 @@
+describe('Runtime Banner Deep Linking', () => {
+  const runtimeStub = {
+    onBeforeLoad: (win) => {
+      Object.defineProperty(win.navigator, 'userAgent', {
+        value:
+          'Mozilla/5.0 (iPhone; CPU iPhone OS 14_4_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Mobile/15E148 Safari/604.1',
+      });
+      Object.defineProperty(win.navigator, 'platform', { value: 'iPhone' });
+    },
+  };
+  beforeEach(() => {
+    cy.testSetup();
+    cy.visit(`/`, runtimeStub);
+  });
+
+  it('should include the correct deep_link param in the banner link', () => {
+    // When visiting the root path the banner should deep link into it
+    cy.get('.runtime-banner > a')
+      .should('have.attr', 'href')
+      .and('contains', `deep_link%3D%2F`);
+    // NOTE: %3D%2F -> '=/' (URL Encoded)
+
+    // When visiting `/tags` the banner should deep link into `/tags`
+    cy.visit('/tags', runtimeStub).then(() => {
+      cy.get('.runtime-banner > a')
+        .should('have.attr', 'href')
+        .and('contains', `deep_link%3D%2Ftags`);
+      // NOTE: %3D%2Ftags -> '=/tags' (URL Encoded)
+    });
+  });
+
+  it('should redirect to custom scheme when someone taps on the banner', () => {
+    const deepLinkPath = '/custom_path';
+    const launchCustomSchemeDeepLinkStub = (url) => {
+      expect(url).to.eq(`com.forem.app:/${deepLinkPath}`);
+    };
+    cy.window()
+      .then((win) => {
+        cy.stub(
```

Comment:
Is stubbing this required or can we use [cy,intercet](https://docs.cypress.io/api/commands/intercept#Usage) instead and handle when the URL is `com.forem.app:/${deepLinkPath}` is hit?

Reply:
I think this is required because when redirecting to custom schemes (like `tel://`) are never accessed (Cypress/browser don't allow it).

[I read about it here](https://medium.com/cypress-io-thailand/understand-stub-spy-apis-that-will-make-your-test-better-than-ever-on-cypress-io-797cb9eb205a#b6ad) and some other sites a few weeks ago and this is why I went with this approach.

I'm not sure if I'm missing something but I just gave it another try using intercept and wasn't able to make it work.